RELEASE-NOTES.md

```diff
@@ -1,5 +1,45 @@
 # Prefect Release Notes
 
+## Release 2.10.6
+
+### Deploy many flows at once with projects
+BP to add detail
+
+### Enhancements
+- Add ability for `flow_run_name` and `task_run_name` settings to accept functions — https://github.com/PrefectHQ/prefect/pull/8933
+- Add pending items count to service failure exception — https://github.com/PrefectHQ/prefect/pull/9306
+- Add "severity" key to JSON formatted logs for GCP compatibility — https://github.com/PrefectHQ/prefect/pull/9200
+- Update orchestration rules to allow transitions from terminal states — https://github.com/PrefectHQ/prefect/pull/9152
+- Update projects to support multiple deployments — https://github.com/PrefectHQ/prefect/pull/9217
+- Add work pool filter for flows — https://github.com/PrefectHQ/prefect/pull/9308
```

Comment:
- Enable filtering flows by work pool at the `/flows/filter` endpoint — [https://github.com/PrefectHQ/prefect/pull/9308]